MSVC: warning C4702: unreachable code

The following error arises when compiling with MSVC (17.5.0 Preview 2.0) with `/W4`:

```
...\dragonbox-src\include\dragonbox\dragonbox.h(662): warning C4702: unreachable code
```

The line in for the warning is: https://github.com/jk-jeon/dragonbox/blob/4b195789bd4c995ea80ce5a65d6c145c70ae9a6b/include/dragonbox/dragonbox.h#L662

However, that isn't the real problem. The warning is actually triggered by the whole second `if constexpr` block at: https://github.com/jk-jeon/dragonbox/blob/4b195789bd4c995ea80ce5a65d6c145c70ae9a6b/include/dragonbox/dragonbox.h#L657

Changing that second `if constexpr` to `else if constexpr` resolves the warning.

--- a/include/dragonbox/dragonbox.h
+++ b/include/dragonbox/dragonbox.h
@@ -654,7 +654,7 @@ namespace jkj::dragonbox {
                 }
                 // Specialize for 64-bit division by 1000.
                 // Ensure that the correctness condition is met.
-                if constexpr (std::is_same_v<UInt, std::uint64_t> && N == 3 &&
+                else if constexpr (std::is_same_v<UInt, std::uint64_t> && N == 3 &&
                               n_max <= std::uint64_t(15534100272597517998ull)) {
                     return wuint::umul128_upper64(n, std::uint64_t(2361183241434822607ull)) >> 7;
                 }

